distributedlog-io/dlfs/src/main/java/org/apache/distributedlog/fs/DLFileSystem.java

```diff
@@ -0,0 +1,330 @@
+                .uri(name)
+                .build();
+        log.info("Initialized the filesystem at {}", name);
+    }
+
+    @Override
+    public void close() throws IOException {
+        // clean up the resource
+        namespace.close();
+        super.close();
+    }
+
+    //
+    // Util Functions
+    //
+
+    private Path makeAbsolute(Path f) {
+        if (f.isAbsolute()) {
+            return f;
+        } else {
+            return new Path(workingDir, f);
+        }
+    }
+
+    private String getStreamName(Path relativePath) {
+        return makeAbsolute(relativePath).toUri().getPath().substring(1);
+    }
+
+    //
+    // Home & Working Directory
+    //
+
+    @Override
+    public Path getHomeDirectory() {
+        return this.makeQualified(new Path(System.getProperty("user.home", "")));
+    }
+
+    protected Path getInitialWorkingDirectory() {
+        return this.makeQualified(new Path(System.getProperty("user.dir", "")));
+    }
+
+    @Override
+    public void setWorkingDirectory(Path path) {
+        workingDir = makeAbsolute(path);
+        checkPath(workingDir);
+    }
+
+    @Override
+    public Path getWorkingDirectory() {
+        return workingDir;
+    }
+
+
+    @Override
+    public FSDataInputStream open(Path path, int bufferSize)
+            throws IOException {
+        try {
+            DistributedLogManager dlm = namespace.openLog(getStreamName(path));
+            LogReader reader;
+            try {
+                reader = dlm.openLogReader(DLSN.InitialDLSN);
+            } catch (LogNotFoundException lnfe) {
+                throw new FileNotFoundException(path.toString());
+            } catch (LogEmptyException lee) {
+                throw new FileNotFoundException(path.toString());
+            }
+            return new FSDataInputStream(
+                new BufferedFSInputStream(
+                    new DLInputStream(dlm, reader, 0L),
+                    bufferSize));
+        } catch (LogNotFoundException e) {
+            throw new FileNotFoundException(path.toString());
+        }
+    }
+
+    @Override
+    public FSDataOutputStream create(Path path,
+                                     FsPermission fsPermission,
+                                     boolean overwrite,
+                                     int bufferSize,
+                                     short replication,
+                                     long blockSize,
+                                     Progressable progressable) throws IOException {
+        // for overwrite, delete the existing file first.
+        if (overwrite) {
+            delete(path, false);
+        }
+
+        DistributedLogConfiguration confLocal = new DistributedLogConfiguration();
+        confLocal.addConfiguration(dlConf);
+        confLocal.setEnsembleSize(replication);
+        confLocal.setWriteQuorumSize(replication);
+        confLocal.setAckQuorumSize(replication);
```

Comment:
ack quorum == write quorum?

Reply:
hdfs api has only one replication factor. so I am making ack quorum == write quorum to match similar behavior as hdfs.